pkg/controllermanager/controller/shoot/shoot_reference_control.go

```diff
@@ -252,6 +314,55 @@ func (r *shootReferenceReconciler) getUnreferencedSecrets(c client.Client, shoot
 	return secretsToRelease, nil
 }
 
+func (r *shootReferenceReconciler) getUnreferencedConfigMaps(c client.Client, shoot *gardencorev1beta1.Shoot) ([]corev1.ConfigMap, error) {
+	namespace := shoot.Namespace
+
+	configMaps := &corev1.ConfigMapList{}
+	if err := r.configMapLister(r.ctx, configMaps, client.InNamespace(namespace)); err != nil {
+		return nil, err
+	}
+
+	// Exit early if there are no ConfigMaps at all in the namespace
+	if len(configMaps.Items) == 0 {
+		return nil, nil
+	}
+
+	shoots := &gardencorev1beta1.ShootList{}
+	if err := c.List(r.ctx, shoots, client.InNamespace(namespace)); err != nil {
+		return nil, err
+	}
+
+	referencedConfigMaps := sets.NewString()
+	for _, s := range shoots.Items {
+		// Ignore own references if shoot is in deletion and references are not needed any more by Gardener.
+		if s.Name == shoot.Name && shoot.DeletionTimestamp != nil && !controllerutils.HasFinalizer(&s, gardencorev1beta1.GardenerName) {
```

Comment:
I think we should not skip shoots in deletion.
As long as the shoot exists, the finalizer on the the config map should not be removed. I see it will be re-added a few moments later in `handleReferencedConfigMap`, but it may be long enough and the garbage collector will delete the configmap.

Reply:
> As long as the shoot exists, the finalizer on the the config map should not be removed

It should be case. Note also the `!controllerutils.HasFinalizer(&s, gardencorev1beta1.GardenerName)` part. It should release the referenced ConfigMap when the `gardener` finalizer is removed from the Shoot (-> the Shoot deletion completed).
Let me try once again this case.

Reply:
Oh, I missed the check for the `gardener` finelizer. Everything is good.

Reply:
I just confirmed one more time - the ConfigMap finalizer is removed after the Shoot deletion completes (its `gardener` finalizer is removed).